packages/widget_toolkit/lib/src/lib_language_picker/views/change_language.dart

```diff
@@ -88,71 +116,92 @@ class LanguagePickerModalConfiguration extends ModalConfiguration {
-  const _ChangeLanguageWidget({
+class ChangeLanguageWidget extends StatelessWidget {
+  const ChangeLanguageWidget({
     required this.messageState,
-    this.errorPanelIcon,
+    required this.service,
+    required this.onChanged,
+    this.itemBuilder,
     this.errorBuilder,
-    this.iconRight,
     Key? key,
   }) : super(key: key);
 
-  final dynamic iconRight;
+  final Widget Function(
+    SelectedLanguageModel model,
+    bool loading,
+    BuildContext context,
+  )? itemBuilder;
   final Widget Function(ErrorModel?)? errorBuilder;
   final MessagePanelState messageState;
-  final dynamic errorPanelIcon;
+  final LanguageService service;
+  final Function(LanguageModel language) onChanged;
 
   @override
-  Widget build(BuildContext context) =>
-      RxBlocBuilder<LanguagePickerBlocType, List<SelectedLanguageModel>>(
-        bloc: context.read<LanguagePickerBlocType>(),
-        state: (bloc) => bloc.states.languages,
-        builder: (context, snapshot, bloc) => Padding(
-          padding: context.languagePickerTheme.changeLanguagePadding,
-          child: SingleChildScrollView(
-            child: Column(
-              children: [
-                MessagePanelError<LanguagePickerBlocType>(
-                  errorState: (bloc) => bloc.states.errors,
-                  padding:
-                      context.languagePickerTheme.messagePanelErrorEdgeInsets,
-                  errorBuilder: errorBuilder,
-                  messageState: messageState,
-                  errorPanelIcon: errorPanelIcon,
-                ),
-                ...(snapshot.data ?? [])
-                    .map(
-                      (language) => _ChooseLanguage(
-                        languageModel: language,
-                        padding:
-                            context.languagePickerTheme.chooseLanguagePadding,
-                        isLoading: language.isLoading,
-                        onPressed: (snapshot.data ?? []).isAnyLoading
-                            ? null
-                            : (languageModel) => context
-                                .read<LanguagePickerBlocType>()
-                                .events
-                                .setCurrent(languageModel.language),
-                        iconRight: iconRight,
+  Widget build(BuildContext context) => MultiProvider(
+        providers: LanguagePickerDependencies.from(
+          service,
+        ).providers,
+        child: Builder(
+          builder: (context) =>
+              RxBlocListener<LanguagePickerBlocType, LanguageModel>(
+            state: (bloc) => bloc.states.currentLanguage,
+            listener: (context, state) => onChanged.call(state),
+            child: RxBlocBuilder<LanguagePickerBlocType,
+                List<SelectedLanguageModel>>(
+              bloc: context.read<LanguagePickerBlocType>(),
+              state: (bloc) => bloc.states.languages,
+              builder: (context, snapshot, bloc) => Padding(
```

Comment:
@StanislavPrime for better readability you can extract the closures as private methods using a method reference

Reply:
The code was extracted in private methods